python3/cloudwatch/src/lambda_function.py

```diff
@@ -72,11 +73,15 @@ def _extract_lambda_log_message(log):
     message_parts = str_message[start_split:].split('\t')
     size = len(message_parts)
     if size == PYTHON_EVENT_SIZE or size == NODEJS_EVENT_SIZE:
-        log['@timestamp'] = message_parts[0]
-        log['requestID'] = message_parts[1]
-        log['message'] = message_parts[size - 1]
-    if size == NODEJS_EVENT_SIZE:
-        log['log_level'] = message_parts[2]
+        try:
+            parser.parse(message_parts[0])
+            log['@timestamp'] = message_parts[0]
+            log['requestID'] = message_parts[1]
+            log['message'] = message_parts[size - 1]
+            if size == NODEJS_EVENT_SIZE:
+                log['log_level'] = message_parts[2]
+        except ValueError:
```

Comment:
1. I am not sure if this will catch the error, I believe it raises a ParseError, and not ValueError. Maybe worth expanding it to every exception and not just specific.
2. Let's add a debug log that prints out the error.